### PR TITLE
Add support for cross-fork issue references

### LINKED ISSUE #17: Support user#1
Cross-fork issues are rendered as `other-user#number`

More info in https://github.com/sindresorhus/linkify-issues/issues/16

I tried adding support in https://github.com/sindresorhus/issue-regex/pull/16 but it's not as easy as making the repo name optional.

### PR DESCRIPTION
**Pull request by fregante**
- Fixes #17 

Blocked by:
- [x]  #18 
- [x] https://github.com/sindresorhus/issue-regex/pull/19

### CODE REVIEW

**Comment by fregante** [comment]
This is failing on patterns like `a/b..#1`

File: test.js [hunk 1]
```diff
@@ -223,12 +223,20 @@ test(
...
 	'forkuser#123',
 	['forkuser',, '123'],
 );
 
+// Source: As of March 2022 GitHub shows an error message when trying to create an organization with a longer name. See issue #11.
+test(
+	matches,
+	'thisorganisationnameistoolongxxxxxxxxxxx/foo#7888',
+	['foo',, '7888'],
+	'GitHub organization names can\'t be longer than 39 characters, so reference should be parsed as forkuser#number',
```
Comment (fregante) [inline test.js:237]: I was initially stuck on this issue.

`IMPOSSIBLE-USERMAME/foo#123` couldn't be linkified, but the `foo#123` part now can be.

Why? The string above is equivalent to `what/ever/string/user#123`, which GitHub successfully linkifies as  `what/ever/string/<user#123>`

**Comment by sindresorhus** [comment]
@fregante Ready for release?

**Comment by fregante** [comment]
Sure. It would unblock the PR in linkify-issues

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/issue-regex/releases/tag/v4.2.0

**Pull request merged**